src/config/timer.js

```diff
@@ -5,109 +5,115 @@ import { Message, MessageWithClickEvent } from "./message.js";
 export class Timer {
   #config;
   #chatComponent;
-	#timeComponent = new TimeComponent();
-	#lastMsgTime = (new Date()).getTime();
+  #timeComponent = new TimeComponent();
+  #lastMsgTime = (new Date()).getTime();
   #currentEventIndex = 0;
 
   constructor(config, chatComponent) {
     this.#config = config;
     this.#chatComponent = chatComponent;
   }
 
-	runClock() {
-		const currentTime = new Date();
-		let currentHour = currentTime.getHours();
-		const currentMinute = currentTime.getMinutes().toString().padStart(2, "0");
-
-		let session = "AM";
-
-		if (currentHour >= 12) {
-			session = "PM";
-		};
-
-		if (currentHour > 12) {
-			currentHour -= 12;
-		};
-
-		currentHour = currentHour.toString().padStart(2, "0");
-
-		const timeHtmlElement = `
-			<p>Today's date: 
-			${currentTime.toLocaleDateString(this.#config.locale, this.#config.dateTimeDisplayOption)}</p>
-			<p>Current time: ${currentHour}:${currentMinute} ${session}</p>
-		`;
-
-		this.#timeComponent.display(timeHtmlElement);
-
-		setTimeout(() => {
-			this.runClock();
-		}, 10000);
-	}
-
-	start() {
-		const endTime = this.#config.endTime; // string "hh:mm:ss"
-
-		if (endTime <= this.getCurrentTime()) {
-			const endDayMsg = new Message(
-				`Hey ${this.#config.userName}, it's the end of day. Great work - you've accomplished a lot today! Get a good rest and see ya 👋`, 
-				this.#config.buddyProfilePicUrl
-			);
-			this.#chatComponent.display(endDayMsg.render());
-			return;
-		};
-
-		const frequencyInMs = this.#config.frequencyInMs;
-		const currentTimeInMs = (new Date()).getTime();
-		const duration = currentTimeInMs - this.#lastMsgTime;
-		if (duration >= frequencyInMs) {
-			if(this.#currentEventIndex >= this.#config.getEventsLength()) {
-				this.#currentEventIndex = 0;
-			};
-			const eventIndex = this.#currentEventIndex++;
-			this.#sendMessage(eventIndex);
-			this.#sendMessageWithClickEvent(eventIndex);
-			this.updateEventMessage();
-			this.#lastMsgTime = currentTimeInMs;
-		};
-
-		setTimeout(() => {
-			this.start();
-		}, 1000);
-	}
-
-	#sendMessage(index) {
+  runClock() {
+    const currentTime = new Date();
+    let currentHour = currentTime.getHours();
+    const currentMinute = currentTime.getMinutes().toString().padStart(2, "0");
+
+    let session = "AM";
+
+    if (currentHour >= 12) {
+      session = "PM";
+    };
+
+    if (currentHour > 12) {
+      currentHour -= 12;
+    };
+
+    currentHour = currentHour.toString().padStart(2, "0");
```

Comment:
Don't overwrite like this.
A variable should only hold one type. If it is a number, don't magically change it to a string.

if you do so, you might forget and cause issues later on